templates/mongodb-shard.conf.erb

```diff
@@ -0,0 +1,21 @@
+<% if @configdb -%>
```

Comment:
In mongos.pp you had the manifest fail if configdb was unset, but in the template you're still checking whether it's set?